python/revng/cli/download_pdb.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import os
+import shutil
+import sys
+
+import pefile
+import requests
+
+from .commands_registry import Command, Options, commands_registry
+from .support import log_error
+
+
+def log(message):
+    sys.stderr.write(message + "\n")
+
+
+def download_pdb_file(url):
+    local_filename = url.split("/")[-1]
+    local_filename = os.path.join("./", local_filename)
+    with requests.get(url, stream=True) as r, open(local_filename, "wb") as f:
+        if r.status_code == 200:
+            shutil.copyfileobj(r.raw, f)
+        else:
+            log("URL was not found.")
+
+    return local_filename
+
+
+class DownloadPDBCommand(Command):
+    def __init__(self):
+        super().__init__(("model", "download-pdb"), "Download missing PDBs for CRTs.")
+
+    def register_arguments(self, parser):
+        parser.add_argument("input", help="The input file.")
+
+    def run(self, options: Options):
+        args = options.parsed_args
+
+        log("Downloading PDB from Microsoft server...")
+        log("Input File: " + args.input)
+        log("")
+
+        if not os.path.exists(args.input):
+            log_error("Could not find " + args.input)
+            return 1
+
+        # Parse the file.
+        try:
+            symbol_server_url = "http://msdl.microsoft.com/download/symbols/"
+            pe = pefile.PE(args.input)
+            for e in pe.DIRECTORY_ENTRY_DEBUG:
+                if not isinstance(e.entry, pefile.Structure):
+                    continue
+
+                pdbfilename = e.entry.PdbFileName.decode("utf-8")
+                # A hash for the PDB file.
+                pdb_id = (
+                    str(hex(e.entry.Signature_Data1)[2:])
+                    + str(hex(e.entry.Signature_Data2)[2:])
+                    + str(hex(e.entry.Signature_Data3)[2:])
+                    + e.entry.Signature_Data4.hex()
+                    + str(hex(e.entry.Age)[2:])
+                )
+                log("PDBID: " + pdb_id)
+
+                pdb_web_location = (
+                    symbol_server_url + pdbfilename + "/" + pdb_id + "/" + pdbfilename
+                )
+                pdb_web_location = pdb_web_location.replace("\x00", "")
+                log("URL: " + pdb_web_location)
+
+                download_pdb_file(pdb_web_location)
+                return 0
+        except Exception as e:
+            log_error("Unable to parse the input file.")
+            log_error(str(e))
+            return 1
```

Comment:
This is a very very large `try` block that catches any exception, including errors such as:
```python
try:
   asd = asd - 3
except:
   pass
```
Restrain the `try` to the very very very bare minimum.